ampd/src/queue/queued_broadcaster.rs

```diff
@@ -300,7 +302,7 @@ mod test {
         assert!(handle.await.unwrap().is_ok());
     }
 
-    #[test]
+    #[test(start_paused = true)]
     async fn should_broadcast_when_gas_limit_has_been_reached() {
```

Comment:
did you run it in a 100+ loop to check that it is stable?

Reply:
Yeah. This is how I tested it:

```
#[test(start_paused = true)]
async fn test_should_broadcast_when_gas_limit_has_been_reached() {
    for _ in 0..5000 {
        should_broadcast_when_gas_limit_has_been_reached().await;
    }
}

async fn should_broadcast_when_gas_limit_has_been_reached() {
    function code.....
}
```